packages/calcite-components/src/utils/dom.ts

```diff
@@ -565,12 +576,25 @@ export function slotChangeHasAssignedElement(event: Event): boolean {
  * ```
  *
  * @param {Event} event The event.
- * @returns {boolean} Whether the slot has any assigned elements.
+ * @param {string} selector The CSS selector string to filter the returned elements by.
+ * @returns {Element[]} An array of elements.
+ */
+export function slotChangeGetAssignedElements(event: Event, selector?: string): Element[] {
+  return getSlotAssignedElements(event.target as HTMLSlotElement, selector);
+}
+
+/**
+ * This helper returns the assigned elements on a `slot` element, filtered by an optional css selector.
+ *
+ * @param {HTMLSlotElement} slot The slot element.
+ * @param {string} selector CSS selector string to filter the returned elements by.
+ * @returns {Element[]} An array of elements.
  */
-export function slotChangeGetAssignedElements(event: Event): Element[] {
-  return (event.currentTarget as HTMLSlotElement).assignedElements({
+export function getSlotAssignedElements<T>(slot: HTMLSlotElement, selector?: string): T[] {
```

Comment:
Generic type should be `T extends Element` and the return type `T[] | null`.

Reply:
Do we really need this method? What is the use case for getting assigned elements from a slot at a time other than an `onSlotChange` event? Wouldn't this kind of promote getting assigned elements at a time other than its really necessary?

To me, it seems like functions like `updateAriaSettings` shouldn't have to get the assigned elements every time. Those elements should be stored during an onSlotChange event for the slot.